### PR TITLE
[Add/edit observation] Clear photo in parent view model

### LINKED ISSUE #1068: Cleared photo reappears on rotating device
**Describe the bug**
The photo response reappears even after removing it manually. This only happens if the screen is rotated after removing the photo.

**To Reproduce**
1. Select any project which contains a photo field
2. Create a new observation and click on add photo
3. Select a new photo 
4. The selected/captured photo should be visible in the edit fragment.
5. Now, remove the photo and rotate the screen.

**Expected behavior**
Removed photo should not reappear

**Actual behavior**
We can see that the photo is visible.

### PR DESCRIPTION
**Pull request by gino-m**
Fixes #1068 (for real this time)

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 1]
```diff
@@ -360,6 +360,10 @@ private void onPhotoResult(Bitmap bitmap) {
     lastPhotoResult.onNext(PhotoResult.create(fieldId, Optional.of(bitmap)));
   }
 
+  public void clearPhoto(String fieldId) {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:363]: @shobhitagarwal1612 Not sure why but this is never called when tapping the trash icon.

**Comment by gino-m** [comment]
@shobhitagarwal1612 Not sure why this isn't working and need to sign off for the day. Any chance you could take a peek?

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 1 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:364]: The top-level was PhotoInputFieldBinding, not PhotoFieldBinding, and I wasn't passing the view model down through the layout vars. Fixed!

**Comment by gino-m** [comment]
This fix works, but is overly complex. Filed #10878 and #1082 for potential cleanup tasks.

@rachaprince Could you PTAL?

**Review by rachaprince — APPROVED** [review]

**Pull request merged**